pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go

```diff
@@ -109,6 +109,10 @@ type KeycloakAPIRealm struct {
 	// Require SSL
 	// +optional
 	SslRequired string `json:"sslRequired,omitempty"`
+
+	// Email
+	// +optional
+	SmtpServer map[string]string `json:"smtpServer,omitempty"`
```

Comment:
Would it not be more user friendly to use a struct with specific fields rather than a map here?  This part of the API is not well documented.  I had to use the admin console and inspect the page to see what parameters were being passed.  I would be nice to have a struct so that it is obvious how to configure this.

Reply:
This crossed my mind too. However, KC [uses a plain map](https://github.com/keycloak/keycloak/blob/f423c0dc5138fe14b985c17b46ff433da4056d25/core/src/main/java/org/keycloak/representations/idm/RealmRepresentation.java#L151) for this too. So it's a question whether we want to keep it consistent with the KC implementation, or improve it at the Operator side.

Reply:
I totally agree with both of you. My impression was that 1:1 was the main goal https://issues.redhat.com/browse/KEYCLOAK-14321 therefore I used the plain map as well.